zproject/backends.py

```diff
@@ -632,13 +632,32 @@ def is_account_control_disabled_user(self, ldap_user: _LDAPUser) -> bool:
         return ldap_disabled
 
     def is_account_realm_access_forbidden(self, ldap_user: _LDAPUser, realm: Realm) -> bool:
-        if "org_membership" not in settings.AUTH_LDAP_USER_ATTR_MAP:
+        if "org_membership" in settings.AUTH_LDAP_USER_ATTR_MAP:
+            org_membership_attr = settings.AUTH_LDAP_USER_ATTR_MAP["org_membership"]
+            allowed_orgs: List[str] = ldap_user.attrs.get(org_membership_attr, [])
+            return not is_subdomain_in_allowed_subdomains_list(realm.subdomain, allowed_orgs)
+        if settings.AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL is not None:
+            realm_access_control = settings.AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL
+            if (
+                isinstance(realm_access_control, dict)
+                and realm.subdomain in realm_access_control
+                and isinstance(realm_access_control[realm.subdomain], dict)
+            ):
+                for attribute in realm_access_control[realm.subdomain]:
+                    if (
+                        attribute in ldap_user.attrs
+                        and realm_access_control[realm.subdomain][attribute]
+                        in ldap_user.attrs[attribute]
+                    ):
+                        return False
+                # If no match is found forbid access
+                return True
+            else:
```

Comment:
Does `AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL = {}` result in no enforcement?  

If so, we might want to have this always be a dictionary with a default of `{}`?

Reply:
I think currently `AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL = {}` would result in not being able to log in.
`AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL = None` and not having `org_membership` set, would result in no enforcement.